cmd/erasure-server-pool-decom.go

```diff
@@ -472,16 +472,23 @@ func (z *erasureServerPools) Init(ctx context.Context) error {
 		// '-1' as argument to decommission multiple pools at a time
 		// but this is not a priority at the moment.
 		for _, pool := range meta.returnResumablePools(1) {
-			go func(pool PoolStatus) {
-				switch err := z.Decommission(ctx, pool.ID); err {
-				case errDecommissionAlreadyRunning:
-					fallthrough
-				case nil:
-					z.doDecommissionInRoutine(ctx, pool.ID)
-				default:
-					logger.LogIf(ctx, fmt.Errorf("Unable to resume decommission of pool %v: %w", pool, err))
-				}
-			}(pool)
+			idx := globalEndpoints.GetPoolIdx(pool.CmdLine)
+			if idx == -1 {
+				return fmt.Errorf("unexpected state present for decommission status pool(%s) not found", pool.CmdLine)
+			}
+			// Only resume decommission on the first node, of the pool under decommission.
+			if globalEndpoints[idx].Endpoints[0].IsLocal {
+				go func(pool PoolStatus) {
+					switch err := z.Decommission(ctx, pool.ID); err {
+					case errDecommissionAlreadyRunning:
```

Comment:
what if decom is running on 2nd node (because 2nd node had received the admin call)? we fall through here and start decom on 1st node as well, right?

Reply:
we do not care @krishnasrinivas where it starts first, upon restart it will start on node 1st of the pool being decommissioned. 

You are basically pointing out a situation where decommission is already running for that we return an error `errDecommissionAlreadyRunning`

Reply:
this check should be 
```
					switch err := z.poolMeta.Decommission(ctx, pool.ID); err {
```

Reply:
> You are basically pointing out a situation where decommission is already running for that we return an error errDecommissionAlreadyRunning

Correct if decom is already running on another node we get  `errDecommissionAlreadyRunning` and the switch statement falls through and calls `z.doDecommissionInRoutine()` on the 1st node. We should not start decom on 1s node if it's already running on another node.

Reply:
@krishnasrinivas that only happens if the returned value is `nil`

Reply:
ok, I see that you have changed the order in the switch cases